app/controllers/application_controller.rb

```diff
@@ -18,8 +18,8 @@ class ApplicationController < ActionController::Base
   before_action :session_expiry, :update_activity_time, :unless => proc {|c| !SETTINGS[:login] || c.remote_user_provided? || c.api_request? }
   before_action :set_taxonomy, :require_mail, :check_empty_taxonomy
   before_action :authorize
```

Comment:
The same applies to `authorize`, the CSP headers must be in place for the 403 page.